sigmastate/src/test/scala/sigmastate/utxo/DistributedSigSpecification.scala

```diff
@@ -0,0 +1,416 @@
+package sigmastate.utxo
+
+import sigmastate._
+import sigmastate.basics.DLogProtocol.DLogInteractiveProver
+import sigmastate.basics.DiffieHellmanTupleInteractiveProver
+import sigmastate.helpers.{ContextEnrichingTestProvingInterpreter, ErgoLikeTestProvingInterpreter, SigmaTestingCommons}
+import sigmastate.interpreter._
+import sigmastate.lang.Terms._
+
+class DistributedSigSpecification extends SigmaTestingCommons {
+
+  implicit lazy val IR: TestingIRContext = new TestingIRContext
+
+  private val ctx = fakeContext
+
+  /**
+    * An example test where Alice (A) and Bob (B) are signing an input in a distributed way. A statement which
+    * protects the box to spend is "pubkey_Alice && pubkey_Bob". Note that a signature in this case is about
+    * a transcript of a Sigma-protocol ((a_Alice, a_Bob), e, (z_Alice, z_Bob)),
+    * which is done in non-interactive way (thus "e" is got via a Fiat-Shamir transformation).
+    *
+    * For that, they are going through following steps:
+    *
+    * - Bob is generating first protocol message a_Bob and sends it to Alice
+    * - Alice forms a hint which contain Bob's commitment "a_Bob", and puts the hint into a hints bag
+    * - She proves the statement using the bag, getting the partial protocol transcript
+    * (a_Alice, e, z_Alice) as a result and sends "a_Alice" and "z_Alice" to Bob.
+    * Please note that "e" is got from both a_Alice and a_Bob.
+    *
+    * - Bob now also knows a_Alice, so can generate the same "e" as Alice. Thus Bob is generating valid
+    * proof ((a_Alice, a_Bob), e, (z_Alice, z_Bob)).
+    */
+  property("distributed AND (2 out of 2)") {
+    val proverA = new ErgoLikeTestProvingInterpreter
+    val proverB = new ErgoLikeTestProvingInterpreter
+    val verifier: ContextEnrichingTestProvingInterpreter = new ContextEnrichingTestProvingInterpreter
+
+    val pubkeyAlice = proverA.dlogSecrets.head.publicImage
+    val pubkeyBob = proverB.dlogSecrets.head.publicImage
+
+    val env = Map("pubkeyA" -> pubkeyAlice, "pubkeyB" -> pubkeyBob)
+    val prop: Values.Value[SSigmaProp.type] = compile(env, """pubkeyA && pubkeyB""").asSigmaProp
+
+    val (rBob, aBob) = DLogInteractiveProver.firstMessage()
+
+    val dlBKnown: Hint = RealCommitment(pubkeyBob, aBob)
```

Comment:
Consider renaming `dlBKnown` to `bobsHint`

Reply:
Actually this is the hint for Alice in the first place, then what does the name mean?

Reply:
by bobsHint, I meant the hint from bob. Maybe its better to call it hintFromBob because dlBKnown seems a bit non-intuitive

Reply:
Renamed to hintFromBob.